src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java

```diff
@@ -109,21 +171,56 @@ private static ArgumentMultimap extractArguments(String argsString, List<PrefixP
         return argMultimap;
     }
 
-    /**
-     * Returns the trimmed value of the argument in the arguments string specified by {@code currentPrefixPosition}.
-     * The end position of the value is determined by {@code nextPrefixPosition}.
-     */
     private static String extractArgumentValue(String argsString,
                                         PrefixPosition currentPrefixPosition,
                                         PrefixPosition nextPrefixPosition) {
         Prefix prefix = currentPrefixPosition.getPrefix();
 
-        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getPrefix().length();
+        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getLength();
+        String value = argsString.substring(valueStartPos, nextPrefixPosition.getStartPosition());
+
+        return value.trim();
+    }
+
+    private static ArgumentMultimap extractFilterArguments(String argsString, List<PrefixPosition> prefixPositions) {
```

Comment:
Maybe you can provide Javadocs for this method to make it easier for developers to understand your logic better

Reply:
Done